src/modules/Instabug.ts

```diff
@@ -0,0 +1,605 @@
+   * set user attributes, or an empty dictionary if no user attributes have been set.
+   */
+  export const getAllUserAttributes = (callback: (attributes: Record<string, string>) => void) => {
+    NativeInstabug.getAllUserAttributes(callback);
+  };
+
+  /**
+   * Clears all user attributes if exists.
+   */
+  export const clearAllUserAttributes = () => {
+    NativeInstabug.clearAllUserAttributes();
+  };
+
+  /**
+   * Enable/Disable debug logs from Instabug SDK
+   * Default state: disabled
+   *
+   * @param isEnabled whether debug logs should be printed or not into LogCat
+   */
+  export const setDebugEnabled = (isEnabled: boolean) => {
+    if (Platform.OS === 'android') {
+      NativeInstabug.setDebugEnabled(isEnabled);
+    }
+  };
+
+  /**
+   * Enables all Instabug functionality
+   * It works on android only
+   */
+  export const enable = () => {
+    if (Platform.OS === 'android') {
+      NativeInstabug.enable();
+    }
+  };
+
+  /**
+   * Disables all Instabug functionality
+   * It works on android only
+   */
+  export const disable = () => {
+    if (Platform.OS === 'android') {
+      NativeInstabug.disable();
+    }
+  };
+
+  /**
+   * Checks whether app is development/Beta testing OR live
+   * Note: This API is iOS only
+   * It returns in the callback false if in development or beta testing on Test Flight, and
+   * true if app is live on the app store.
+   * @param callback callback with argument as return value 'isLive'
+   */
+  export const isRunningLive = (callback: (isLive: boolean) => void) => {
+    if (Platform.OS === 'ios') {
+      NativeInstabug.isRunningLive(callback);
+    }
+  };
+
+  /**
+   * Shows the welcome message in a specific mode.
+   * @param mode An enum to set the welcome message mode to live, or beta.
+   */
+  export const showWelcomeMessage = (mode: ArgsRegistry.welcomeMessageMode) => {
+    NativeInstabug.showWelcomeMessageWithMode(mode);
+  };
+
+  /**
+   * Sets the welcome message mode to live, beta or disabled.
+   * @param mode An enum to set the welcome message mode to live, beta or disabled.
+   */
+  export const setWelcomeMessageMode = (mode: ArgsRegistry.welcomeMessageMode) => {
+    NativeInstabug.setWelcomeMessageMode(mode);
+  };
+
+  /**
+   * Add file to be attached to the bug report.
+   * @param filePath
+   * @param fileName
+   */
+  export const addFileAttachment = (filePath: string, fileName: string) => {
+    if (Platform.OS === 'android') {
+      NativeInstabug.setFileAttachment(filePath, fileName);
+    } else {
+      NativeInstabug.setFileAttachment(filePath);
+    }
+  };
+
+  /**
+   * @deprecated Use {@link addPrivateView} instead.
+   *
+   * Hides component from screenshots, screen recordings and view hierarchy.
+   * @param viewRef the ref of the component to hide
+   */
+  export const setPrivateView = (viewRef: number | React.Component | React.ComponentClass) => {
```

Comment:
It might a bit ugly, and also verbose. But looking into that this a public API, having some type verbosity gives the user a better readability with their editor intellisense features, without the need to dig deeper to know what type is needed.

Consider we created a type called `ComponentHandler`, now `viewRef` would be of this new type, hovering over the function still doesn't give any tips about what this is. On the other hand, if we have a verbose type (as the current one), the user will just know what they need to pass.

Reply:
Sounds reasonable.
One note though, this type doesn't seem to be correct, I think this method takes a ref object not a component or a number. Shouldn't it be of type `React.RefObject<T>`?

Reply:
You can find the correct type [here](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/ae199c4d1ac55cbc848449934d3cc59bc3ea9918/types/react-native/index.d.ts#L9812-L9814).